src/combinatorics/derangements/exponential.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2021 Henry Swanson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henry Swanson, Patrick Massot
+-/
+import combinatorics.derangements.finite
+import analysis.complex.basic
+import data.complex.exponential
+import topology.metric_space.cau_seq_filter
+
+/-!
+# Derangement exponential series
+
+This file proves that the probability of a permutation on n elements being a derangement is 1/e.
+The specific lemma is `num_derangements_tendsto_inv_e`.
+-/
+open filter
+open finset
+
+open_locale big_operators
+open_locale topological_space
+
+lemma complex.tendsto_iff_real (u : ℕ → ℝ) (x : ℝ) :
+  tendsto (λ n, u n) at_top (𝓝 x) ↔
+  tendsto (λ n, (u n : ℂ)) at_top (𝓝 (x : ℂ)) :=
+  ⟨(complex.continuous_of_real.tendsto x).comp, (complex.continuous_re.tendsto x).comp⟩
+
+lemma complex.tendsto_exp_series (z : ℂ) :
+  tendsto (λ n, ∑ k in range n, z^k / k.factorial) at_top (𝓝 z.exp) :=
+begin
+  convert z.exp'.tendsto_limit,
+  unfold complex.exp,
+end
+
+lemma real.tendsto_exp_series (x : ℝ) :
+  tendsto (λ n, ∑ k in range n, x^k / k.factorial) at_top (𝓝 x.exp) :=
+begin
+  rw complex.tendsto_iff_real,
+  convert complex.tendsto_exp_series x; simp,
+end
+
+theorem num_derangements_tendsto_inv_e :
+  tendsto (λ n, (num_derangements n : ℝ) / n.factorial) at_top
+  (𝓝 (real.exp (-1))) :=
```

Comment:
Could this be stated using `has_sum` rather than `tendsto`?

Reply:
Not easily, I don't think. The `num_derangements` function isn't defined as a partial series, so I think it'd be harder to prove `has_sum` for `num_derangements (n+1) - num_derangements n`.

Reply:
Oh, I see, yes.